WordPress/Classes/ViewRelated/Blog/Site Settings/HomepageSettingsViewController.swift

```diff
@@ -118,37 +144,79 @@ import WordPressShared
             postsPageTitle = postsPage.titleForDisplay()
         }
 
-        let selectPage: ImmuTableAction = { _ in
+        let homepageRow: ImmuTableRow
+        if case .selectedHomepage = updateState {
+            homepageRow = ActivityIndicatorRow(title: Strings.homepage, animating: true, action: nil)
+        } else {
+            homepageRow = NavigationItemRow(title: Strings.homepage, detail: homepageTitle, action: { _ in
+                self.pushPageSelection(selectedPostID: self.blog?.homepagePageID) { [weak self] selected in
+                    if let postID = selected.postID?.intValue {
+                        self?.setHomepageType(.page, postsPageID: self?.blog.homepagePostsPageID, homePageID: postID)
+                    }
+                }
+            })
         }
 
-        let homepageRow = NavigationItemRow(title: Strings.homepage, detail: homepageTitle, action: selectPage)
-        let postsPageRow = NavigationItemRow(title: Strings.postsPage, detail: postsPageTitle, action: selectPage)
+        let postsPageRow: ImmuTableRow
+        if case .selectedPostsPage = updateState {
+            postsPageRow = ActivityIndicatorRow(title: Strings.postsPage, animating: true, action: nil)
+        } else {
+            postsPageRow = NavigationItemRow(title: Strings.postsPage, detail: postsPageTitle, action: { _ in
+                self.pushPageSelection(selectedPostID: self.blog?.homepagePostsPageID) { [weak self] selected in
+                    if let postID = selected.postID?.intValue {
+                        self?.setHomepageType(.page, postsPageID: postID, homePageID: self?.blog.homepagePageID)
+                    }
+                }
+            })
+        }
 
         return [homepageRow, postsPageRow]
     }
 
-    // MARK: - Updating Settings
+    open override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        animateDeselectionInteractively()
+    }
 
-    fileprivate func startUpdating() {
-        updating = true
+    fileprivate func pushPageSelection(selectedPostID: Int?, _ completion: @escaping (Page) -> Void) {
+        let viewController = SelectPostViewController(blog: blog,
+                                                      isSelectedPost: { $0.postID?.intValue == selectedPostID },
+                                                      showsPostType: false,
+                                                      entityName: Page.entityName()) { (post) in
```

Comment:
Can we keep the trailing closure name? I think it might just be a little easier to understand what the closure's for.